src/modules/sharepoint/sharepoint.service.ts

```diff
@@ -1,24 +1,213 @@
-import { Injectable } from '@nestjs/common';
+import { UploadResult } from '@microsoft/microsoft-graph-client';
+import { Inject, Injectable } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import SharepointConfig from '@ukef/config/sharepoint.config';
+import { SharepointResourceTypeEnum } from '@ukef/constants/enums/sharepoint-resource-type';
 import GraphService from '@ukef/modules/graph/graph.service';
 
+import { GraphCreateSiteResponseDto } from '../graph/dto/graph-create-site-response.dto';
 import { ListItem } from './list-item.interface';
+import { AndListItemFilter } from './list-item-filter/and.list-item-filter';
+import { FieldEqualsListItemFilter } from './list-item-filter/field-equals.list-item-filter';
+import { FieldNotNullListItemFilter } from './list-item-filter/field-not-null.list-item-filter';
 import { ListItemFilter } from './list-item-filter/list-item-filter.interface';
 
+export interface SharepointGetResourcesParams {
+  ukefSiteId: string;
+  sharepointResourceType: SharepointResourceTypeEnum;
+}
+
+export interface SharepointGetItemsParams {
+  ukefSiteId: string;
+  listId: string;
+}
+
+export interface SharepointCreateSiteParams {
+  exporterName: string;
+  newSiteId: string;
+}
+
+export interface SharepointGetDealFolderParams {
+  siteId: string;
+  dealFolderName: string;
+}
+
+export interface SharepointGetBuyerFolderParams {
+  siteId: string;
+  buyerName: string;
+}
+
+export interface SharepointUploadFileParams {
+  file: NodeJS.ReadableStream;
+  fileSizeInBytes: number;
+  fileName: string;
+  urlToCreateUploadSession: string;
+}
+
+export interface SharepointFindListItems<Fields> {
+  siteUrl: string;
+  listId: string;
+  fieldsToReturn: (keyof Fields)[];
+  filter: ListItemFilter;
+}
+
 @Injectable()
 export class SharepointService {
-  constructor(private readonly graphService: GraphService) {}
+  constructor(
+    private readonly graphService: GraphService,
+    @Inject(SharepointConfig.KEY)
+    private readonly sharepointConfig: ConfigType<typeof SharepointConfig>,
+  ) {}
+
+  async getSiteByUkefSiteId(ukefSiteId: string): Promise<{ id: string }> {
+    return await this.graphService.get<{ id: string }>({
```

Comment:
I think we can make all the `return await`s in the file just `return` instead (and then remove the `async` modifier as well)

Reply:
This will be solved in a later PR